batch/batch/cloud/gcp/instance_config.py

```diff
@@ -0,0 +1,114 @@
+from typing import List
+
+from ...instance_config import InstanceConfig, is_power_two, QuantifiedResource
+from .resource_utils import gcp_machine_type_to_parts
+
+
+GCP_INSTANCE_CONFIG_VERSION = 4
+
+
+class GCPSlimInstanceConfig(InstanceConfig):
+    def __init__(self,
+                 machine_type: str,
+                 preemptible: bool,
+                 local_ssd_data_disk: bool,
+                 data_disk_size_gb: int,
+                 boot_disk_size_gb: int,
+                 job_private: bool,
+                 ):
+        self.cloud = 'gcp'
+        self._machine_type = machine_type
+        self.preemptible = preemptible
+        self.local_ssd_data_disk = local_ssd_data_disk
+        self.data_disk_size_gb = data_disk_size_gb
+        self.job_private = job_private
+        self.boot_disk_size_gb = boot_disk_size_gb
+
+        machine_type_parts = gcp_machine_type_to_parts(self._machine_type)
+        assert machine_type_parts is not None, machine_type
+        self._instance_family = machine_type_parts.machine_family
+        self._worker_type = machine_type_parts.worker_type
+        self.cores = machine_type_parts.cores
+
+    def worker_type(self) -> str:
+        return self._worker_type
+
+    @staticmethod
+    def from_dict(data: dict) -> 'GCPSlimInstanceConfig':
+        if data['version'] < 4:
+            disks = data['disks']
+            assert len(disks) == 2
+            assert disks[0]['boot']
+            boot_disk_size_gb = disks[0]['size']
+            assert not disks[1]['boot']
+            local_ssd_data_disk = disks[1]['type'] == 'local-ssd'
+            data_disk_size_gb = disks[1]['size']
+            job_private = data['job-private']
+            return GCPSlimInstanceConfig(
+                data['vm_config']['machineType'],
```

Comment:
I'm not sure this is correct. I think `vm_config` is something I added in this PR and didn't exist before. I'm not 100% sure though.

Reply:
Yeah,  you're right this isn't present.